src/unix/tcp.c

```diff
@@ -198,11 +201,50 @@ int uv__tcp_bind(uv_tcp_t* tcp,
 }
 
 
+static int uv__is_ipv6_link_local(const struct sockaddr* addr) {
+  const struct sockaddr_in6* a6;
+  uint8_t b[2];
+
+  if (addr->sa_family != AF_INET6)
+    return 0;
+
+  a6 = (const struct sockaddr_in6*) addr;
+  memcpy(b, &a6->sin6_addr, sizeof(b));
+
+  return b[0] == 0xFE && b[1] == 0x80;
```

Comment:
Is this `memcpy()` actually needed? Can't we access those bytes directly?

Reply:
It's because I'm not 100% sure the field is defined as `unsigned char s6_addr[16]` everywhere.

If it has a different name, it won't compile, and if it's (signed) `char` the implicit sign extension conversion to `int` produces the wrong result.

Reply:
Got it. I was thinking of smthg like this:
```c
uint8_t* b;

b = (uint8_t*) &a6->sin6_addr;
```

Reply:
In a very literal reading of the language spec that's UB because only `char` is allowed to alias and `uint8_t` isn't `char`. `memcpy()` on the other hand is always safe.